docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
+is passed to the hypervisor and used to boot the VM.
+The default kernel provided in Kata Containers is highly optimized for
+kernel boot time and minimal memory footprint, providing only those
+services required by a container workload. This is based on a very
+current upstream Linux kernel.
 
 ### Guest image
 
-Kata Containers supports both an `initrd` and `rootfs` based minimal guest image.
+The hypervisor uses an image file which provides a minimal root
+filesystem used by the guest kernel to boot the VM and host the Kata Container.
+
+Kata Containers supports both initrd and rootfs based minimal guest images.
+
+The default images are created using the [`osbuilder`](../../tools/osbuilder) tool.
+
+> **Note:**
+>
+> The guest image is *unrelated* to the image used in a container
+> workload.
+>
+> For example, if a user creates a container that runs a shell in a
+> BusyBox image, they will run that shell in a BusyBox environment.
+> However, the guest image running inside the VM that is used to
+> *host* that BusyBox image could be running Clear Linux, Ubuntu,
+> Fedora or any other distribution potentially.
+>
+> The `osbuilder` tool provides
+> [configurations for various common Linux distributions](../../tools/osbuilder/rootfs-builder)
+> which can be built into either initrd or rootfs guest images.
 
 #### Root filesystem image
 
-The default packaged root filesystem image, sometimes referred to as the "mini O/S", is a
-highly optimized container bootstrap system based on [Clear Linux](https://clearlinux.org/). It provides an extremely minimal environment and
-has a highly optimized boot path.
+The default packaged root filesystem image ("rootfs"), sometimes
+referred to as the "mini O/S", is a highly optimized container
+bootstrap system based on [Clear Linux](https://clearlinux.org). It
+provides an extremely minimal environment and has a highly optimized
+boot path.
+
+The only services running in the context of the mini O/S are:
+
+| Process | Environment | systemd service? | Notes |
+|-|-|-|-|
+| systemd | VM root | n/a | The init daemon, running as PID 1 |
+| [Agent](#agent) | VM root | yes | Runs as a systemd service |
+| `chronyd` | VM root | yes | Used to synchronise the time with the host |
+| container workload | VM container | no | Managed by the agent |
+
+See also the [process overview](#process-overview).
 
-The only services running in the context of the mini O/S are the init daemon
-(`systemd`) and the [Agent](#agent). The real workload the user wishes to run
-is created using libcontainer, creating a container in the same manner that is done
-by `runc`.
+> **Note:** The container workload is running inside a full container
+> environment which itself is running within a VM environment.
 
-For example, when `ctr run -ti ubuntu date` is run:
+For example, when the user runs `ctr run -ti ubuntu date`:
 
 - The hypervisor will boot the mini-OS image using the guest kernel.
```

Comment:
I know this wasn't part of your review but I think adding a step above that says "Kata will launch the configured hypervisor" before the hypervisor does the boot helps connect the 12 steps listed above with these condensed steps.